### DISCUSSION TITLE
NodeJS Client working with Date

### DISCUSSION THREAD

**Discussion by asapostolov** [ravendb/ravendb#13239] [Q&A] [answered]
The NodeJS article about how RavenDB is working with dates uses C# `DateTime` and `DateTimeOffset` which is not helpful in the node context

https://ravendb.net/docs/article-page/5.3/nodejs/client-api/how-to/store-dates

 It would be great to have that page localized for NodeJS / TypeScrpt

**Comment by ayende** [comment, marked as answer]
You are correct, and this is tracked here: https://issues.hibernatingrhinos.com/issue/RDoc-2032

In general, you should be able to just work with `Date` objects and it will do the right thing for you.

**Comment by asapostolov** [comment]
Thank you, @ayende. I would also add that the NodeJS index definitions could be improved, right now it's very user unfriendly imo.

```
class Employees_ByFirstAndLastName extends AbstractIndexCreationTask {
    constructor() {
        super();
        this.map =  "docs.Employees.Select(employee => new {" +
            "    FirstName = employee.FirstName," +
            "    LastName = employee.LastName" +
            "})";
    }
}
```
source https://ravendb.net/docs/article-page/4.2/nodejs/indexes/what-are-indexes

Another suggestion - I would also love to have examples for Typescript as well as Javascript in the docs. You can implement that as another tab besides NodeJS/RQL ones. e.g. `NodeJS Javascript` / `NodeJS Typescript` / `RQL`.

**Reply by asapostolov** [reply]
I can see that there's an actual [better syntax for Typescript](https://ravendb.net/articles/ravendb-node-js-typescript-client-5-0-released) but it's not documented anywhere, so I have to figure out how to make it work. Nevertheless - it's awesome that you're putting effort into this!

**Reply by ayende** [reply]
Yes, that is a missing item. I created this: https://issues.hibernatingrhinos.com/issue/RDoc-2033

We'll work on that soon

**Comment by ml054** [comment]
@asapostolov About JS indexes syntax. 

There is strongly typed syntax available, see tests here: https://github.com/ravendb/ravendb-nodejs-client/blob/3e1b1d3ef559e58523714a4224ead5ca9a3d2db9/test/Ported/Indexing/JavaScriptIndexTest.ts

In docs - you are right we're using old syntax, which will be updated.
